### PR TITLE
WIP: Add wallet tool test

### PR DESCRIPTION
**Pull request by promag**
Based on #13926, this PR adds a test for the wallet tool.

### CODE REVIEW

**Comment by jnewbery** [comment]
Thanks @promag - I'll review this later.

File: src/wallet/wallettool.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            return false;
+        }
+
+        CWallet dummyWallet("dummy", WalletDatabase::CreateDummy());
+        std::string backup_filename;
+        if (!WalletBatch::Recover(path, (void*)&dummyWallet, WalletBatch::RecoverKeysOnlyFilter, backup_filename)) {
+            fprintf(stderr, "Salvage failed\n");
+            return false;
+        }
+        //TODO, set wallets best block to genesis to enforce a rescan
```
Comment (practicalswift) [inline src/wallet/wallettool.cpp]:
```
2018-09-22 20:39:40 cpplint(pr=14283): src/wallet/wallettool.cpp:125:  Should have a space between // and comment  [whitespace/comments] [4]
```

File: test/functional/tool_wallet.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+class ToolWalletTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def process(self, *args):
+        binary = self.config["environment"]["BUILDDIR"] + '/src/bitcoin-wallet-tool' + self.config["environment"]["EXEEXT"]
+        input = None
```
Comment (practicalswift) [inline test/functional/tool_wallet.py]: Remove this line. It redefines the built-in function `input`. Furthermore it is unused :-)

**Comment by jnewbery** [comment]
Looks good. Thanks @promag! I've cherry picked and squashed your commits into #13926 . I'll add my review comments there.

**Pull request closed without merging**